app/main.py

```diff
@@ -116,3 +117,25 @@ async def generate_llama(request: OllamaGenerateRequest):
 
     generated_text = response.get("response", "")
     return {"generated_text": generated_text}
+
+@app.post("/llm_tags_summary")
+async def generate_tags_summary(request: OllamaGenerateRequest):
+    """
+    Endpoint to generate tags and summary using Llama 3.1 model (and other models) via the Ollama API.
+    """
+    print(request)
+    if not request.prompt:
```

Comment:
Metadata request is different than the main request with prompt.

Should be executed after the main LLM call on top of its results

It should use metadata prompt defined in env variable for configuration purposes

Then it should always return JSON object - I proposed its structure in the other comment